doc/source/examples/incompressible-flow/2d-sudden-expansion-flow/2d-sudden-expansion-flow.rst

```diff
@@ -0,0 +1,332 @@
+        set number of fluids = 1
+        subsection fluid 0
+            set kinematic viscosity = 1.42857e-4 # Re_h=u*h/nu
+        end
+    end
+
+The viscosity :math:`\nu = 1.42857 \times 10^{-4} m^2/s` corresponds to :math:`\textrm{Re} = 70`. For :math:`\textrm{Re} = 610`, that should be changed to :math:`\nu = 1.6393 \times 10^{-5} m^2/s`.
+
+Mesh
+~~~~
+
+The mesh is composed by bilinear quadrilateral elements, and the problem geometry was taken from Durst et. al. (1993) [#durst1993]_. The inlet dimension was chosen to be large enough to for the flow to fully develop before the channel expansion.
+
+.. code-block:: text
+
+    subsection mesh
+      set type      = gmsh
+      set file name = ./sudden-expansion-flow.msh
+    end
+
+Mesh Adaptation
+~~~~~~~~~~~~~~~
+
+The mesh adaptation algorithm is based on the Kelly error estimator applied on the velocity variable. For :math:`\textrm{Re} = 70` the correspondent code section looks as follows:
+
+.. code-block:: text
+
+    subsection mesh adaptation
+      set variable             = velocity
+      set type                 = kelly
+      set fraction refinement  = 0.2
+      set fraction coarsening  = 0
+      set frequency            = 1
+      set min refinement level = 0
+      set max refinement level = 8
+    end
+
+In this case, the algorithm thoroughly discretizes the mesh around the expansion section, where the recirculation zones appear:
+
+.. image:: image/mesh-adaptation-Re70.png
+
+For :math:`\textrm{Re} = 610`, mesh adaptation was limited by the maximum refinement level, the fraction refinement, and the number of cells, so that the mesh discretization did not became too computationally expensive at the final simulation time. 
+
+Also, it is useful not to have a very refined mesh at the beginning of the simulation (when the flow is still being developed at the inlet channel) and rather allow the algorithm to allocate more cells as the flow becames turlent on the outlet section.
+
+.. code-block:: text
+
+    subsection mesh adaptation
+      set variable                    = velocity
+      set type                        = kelly
+      set fraction refinement         = 0.05
+      set fraction coarsening         = 0
+      set frequency                   = 5
+      set min refinement level        = 0
+      set max refinement level        = 2
+      set max number elements         = 250000
+      set mesh refinement controller  = true
+    end
+
+FEM
+~~~
+
+A linear interpolation order is chosen for the velocity and pressure fields for both :math:`\textrm{Re}` values:
+
+.. code-block:: text
+
+    subsection FEM
+      set pressure order = 1
+      set velocity order = 1
+    end
+
+Boundary Conditions
+~~~~~~~~~~~~~~~~~~~
+
+The inlet velocity is prescribed on boundary 0 as :math:`u = [1, 0, 0]`, and boundary 1 has a prescribed pressure of :math:`p=0` -- identified as ``outlet`` in Lethe. The upper and lower walls (ID 2) have a no-slip Dirichlet boundary condition.
```

Comment:
Be careful, the p=0 boundary condition is actually a "zero traction" boundary condition. like a do-nothing boundary condition.
The reason why we call it an outlet is that if the flow tries to come back, we weakly enforce it to become 0.